### ISSUE TITLE
How can I put edge weight information to the loss ?

### ISSUE THREAD

**Issue by jindeok** [rusty1s/pytorch_geometric#2776]
Hello, Thanks for your great supports. I have an obstacle in implementing GNN with edge-weight though..

![image](https://user-images.githubusercontent.com/35905280/123516381-c1efdd80-d6d6-11eb-8f38-a80a9efbcea8.png)

Upper capture is my GNN with GraphConv layers, 
I want to reflect edge weight to my unsupervised loss function, 
![image](https://user-images.githubusercontent.com/35905280/123516951-ab975100-d6d9-11eb-9547-773fdfa342da.png)
this is the current version and what I want to implement is the formulation below...
![image](https://user-images.githubusercontent.com/35905280/123516903-77239500-d6d9-11eb-8abf-f94c871fda42.png)
I want to reflect edge weight in loss function, and now I am suffering from how to deal with it.
thanks for reading this.

**Comment by rusty1s** [comment]
Given that you have a sparse matrix holding the values `e_{u,v}` (e.g., in a `torch_sparse.SparseTensor), you should be able to do do:
```python
n_id, pos_n_id, _ = n_id.split(n_id.size(0) // 3)

... = (edge_weight[n_id, pos_n_id] * out * pos * out).sum(-1)
```
